Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -314,7 +319,12 @@ calculateFinalValuesForFailedDistanceCalculations handle@ServiceHandle {..} book
   distanceDiff <- getDistanceDiff booking approxTraveledDistance
 
   if distanceDiff < 0
-    then recalculateFareForDistance handle booking ride (approxTraveledDistance + roundToIntegral (abs (distanceDiff * 0.5)))
+    then do
+      let recalculationDistance =
+            min
+              (approxTraveledDistance + roundToIntegral (abs (distanceDiff * 0.5)))
+              (approxTraveledDistance + (50 - fromIntegral (fromMaybe 0 booking.fareParams.driverSelectedFare)))
```

Comment:
The logic is to calculate fare based on both logics and do a min on fare, not distance.
Here the 50 - Driver Added Fare is a currency value, which cannot be added to a distance value.